src/Http.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Scripted;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\ClientException;
+use JmesPath;
+use Scripted\Exceptions\AccessTokenIsUnauthorized;
+
+/**
+ *
+ */
+class Http
+{
+    /**
+     * Default cache length in seconds.
+     *
+     * @var integer
+     */
+    public const DEFAULT_CACHE_LENGTH_SECONDS = 600;
+
+
+    /**
+     * Http get verb.
+     *
+     * @var string
+     */
+    public const GET = 'GET';
+
+    /**
+     * Http post verb.
+     *
+     * @var string
+     */
+    public const POST = 'POST';
+
+
+    /**
+     * Attempts to make an HTTP request with the given parameters.
+     *
+     * @param  string  $path
+     * @param  string $verb
+     * @param  array  $config
+     *
+     * @return mixed
+     * @throws AccessTokenIsUnauthorized
+     *
+     * @see https://github.com/jmespath/jmespath.php
+     * @see http://docs.guzzlephp.org/en/stable/
+     */
+    public static function getResponse($path, $verb, array $config = array())
+    {
+        $orgKey = JmesPath\search('orgKey', $config) ?: Config::getOrgKey();
+        $accessToken = JmesPath\search('accessToken', $config) ?: Config::getAccessToken();
+
+        if (!$orgKey || !$accessToken) {
+            throw new AccessTokenIsUnauthorized();
+        }
+
+        $clearCache = (bool) JmesPath\search('clearCache', $config);
+        $url = sprintf(
+            '%s/%s/v1/%s',
+            Config::BASE_API_URL,
+            $orgKey,
+            $path
+        );
+        $cacheKey = sprintf('%s::%s::%s', $orgKey, $accessToken, $url);
+
+        if ($clearCache) {
+            WordPressApi::setCache($cacheKey, null);
+        }
+
```

Comment:
Not sure if you'd benefit from longer caching here, but see https://wordpress.stackexchange.com/questions/284177/any-reason-why-wp-cache-set-not-to-work

Specifically:
`
The wp_cache_*() functions are non-persistent caches which means that they only last during the current page request....If you want to save these through multiple page page loads but also expire them eventually, you may want to save them as a transient. They work the same way as the WP Object Cache but they are saved to the database for a set amount of time.`

Reply:
Same comment applies to all instances of `WordPressApi::setCache`

Reply:
Can you say more about this? The goal is to leverage persistent caching IF the WordPress instance [has support for it](https://codex.wordpress.org/Class_Reference/WP_Object_Cache#Persistent_Cache_Plugins). My understanding is that `wp_cache_get` will take care of that for me... 

https://github.com/Scripted/wordpress-plugin/blob/5d56422c05412010550392e726baaa85321c1cbd/src/WordPressApi.php#L48-L59

https://github.com/Scripted/wordpress-plugin/blob/5d56422c05412010550392e726baaa85321c1cbd/src/WordPressApi.php#L277-L289

Based on that goal, is this the best strategy? If not, what is the best way to proceed here?

Reply:
Yeah it looks like you're right about that. From the codex article that you linked:

>If persistent caching is configured, then...functions will use the wp_cache functions described in this document. However if persistent caching has not been enabled, then the data will instead be cached to the options table.

So it seems like you should be all set. 👍